src/Microsoft.Windows.CsWin32/MetadataIndex.cs

```diff
@@ -155,7 +155,7 @@ void PopulateNamespace(NamespaceDefinition ns, string? parentNamespace)
                                     fieldEnum.MoveNext();
                                     FieldDefinitionHandle fieldHandle = fieldEnum.Current;
                                     FieldDefinition fieldDef = mr.GetFieldDefinition(fieldHandle);
-                                    if (fieldDef.DecodeSignature(SignatureHandleProvider.Instance, null) is PrimitiveTypeHandleInfo { PrimitiveTypeCode: PrimitiveTypeCode.IntPtr or PrimitiveTypeCode.UIntPtr })
+                                    if (fieldDef.DecodeSignature(metadataFile.Generator.SignatureHandleProvider, null) is PrimitiveTypeHandleInfo { PrimitiveTypeCode: PrimitiveTypeCode.IntPtr or PrimitiveTypeCode.UIntPtr })
```

Comment:
As this appears to be the only reason we need `Generator` on the `MetadataFile`, hopefully this will be an easy coupling to undo.
Given we're only interested in _primitive_ type handles, which have no affinity to a particular winmd and therefore shouldn't need a particular `Generator`, and yet a full fledged `SignatureHandleProvider` *does* have a legit need to be affinitized to a particular `Generator`, I wonder if the solution is to create a special `SignatureHandleProvider`-like class that is just for indexes, and it will only support `PrimitiveTypeHandleInfo` since that's all we need. If the signature is _not_ such a handle, it can return null or whatever else is convenient. (We wouldn't want to _throw_ since this is a hot path.)

Reply:
Thanks. I added a PrimitiveSignatureHandleGenerator and that seems to have reduced the fallout from the change. Still running tests but looks good so far.